Fix apk package pinning on Dockerfile

### Actual behavior

Unable to build Docker images for the blockchain-connector and gateway microservices.

```
=> ERROR [builder 2/5] RUN apk add --no-cache alpine-sdk=~1.0-r1 python3=~3.10.10-r0 cmake=~3.24.4-r0 &&     adduser -D builder &&     mkdir /home/builder/build &&     chown -R builder:builder /home/builder/                                                                                                                                                                                            0.7s
------                                                                                                                                                                                                                                                                                                                                                                                                           
 > [builder 2/5] RUN apk add --no-cache alpine-sdk=~1.0-r1 python3=~3.10.10-r0 cmake=~3.24.4-r0 &&     adduser -D builder &&     mkdir /home/builder/build &&     chown -R builder:builder /home/builder/:                                                                                                                                                                                                       
#9 0.171 fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/main/aarch64/APKINDEX.tar.gz                                                                                                                                                                                                                                                                                                                          
#9 0.415 fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/community/aarch64/APKINDEX.tar.gz                                                                                                                                                                                                                                                                                                                     
#9 0.648 ERROR: unable to select packages:                                                                                                                                                                                                                                                                                                                                                                       
#9 0.666   python3-3.10.11-r0:
#9 0.666     breaks: world[python3~3.10.10-r0]
------
executor failed running [/bin/sh -c apk add --no-cache alpine-sdk=~1.0-r1 python3=~3.10.10-r0 cmake=~3.24.4-r0 &&     adduser -D builder &&     mkdir /home/builder/build &&     chown -R builder:builder /home/builder/]: exit code: 1
make: *** [build-connector] Error 1
```

### Expected behavior

Should always be able to build images Dockerfile for all the microservices

### Steps to reproduce

- make build

### Which version(s) does this affect? (Environment, OS, etc...)

v0.7.0-alpha.12

--- a/services/blockchain-connector/Dockerfile
+++ b/services/blockchain-connector/Dockerfile
@@ -2,7 +2,7 @@
 
 FROM node:16-alpine AS builder
 
-RUN apk add --no-cache alpine-sdk=~1.0-r1 python3=~3.10.10-r0 cmake=~3.24.4-r0 && \
+RUN apk add --no-cache alpine-sdk=~1.0 python3=~3.10 cmake=~3.24 && \
     adduser -D builder && \
     mkdir /home/builder/build && \
     chown -R builder:builder /home/builder/

--- a/services/gateway/Dockerfile
+++ b/services/gateway/Dockerfile
@@ -19,7 +19,7 @@ RUN npm ci && \
 
 FROM node:16-alpine
 
-RUN apk add --no-cache curl=~7.88.1-r1 && \
+RUN apk add --no-cache curl=~8.0 && \
     adduser -D lisk
 
 USER lisk

